src/client/dfs/dfs.c

```diff
@@ -824,7 +824,8 @@ open_symlink(dfs_t *dfs, daos_handle_t th, dfs_obj_t *parent, int flags,
 	int			rc;
 
 	if (flags & O_CREAT) {
-		if (value == NULL || strnlen(value, PATH_MAX-1) > PATH_MAX-1)
+		if (value == NULL ||
+		    strnlen(value, PATH_MAX - 1) == PATH_MAX - 1)
 			return -DER_INVAL;
```

Comment:
This is not part of this patch. I think that we are mixing errno codes and DER_ codes on this function

`return daos_der2errno(-DER_INVAL)`
or
`return EINVAL;`

Reply:
yes you are right.